airbyte-config/models/src/main/java/io/airbyte/config/helpers/CloudLogs.java

```diff
@@ -51,8 +51,35 @@
   List<String> tailCloudLog(Configs configs, String logPath, int numLines) throws IOException;
 
   /**
-   * @return true if configuration is not set
+   * @return true if no cloud logging configuration is set;
    */
-  boolean hasEmptyConfigs(Configs configs);
+  static boolean hasEmptyConfigs(Configs configs) {
```

Comment:
This is not a great interface to pass `Configs`. You're passing too much data and with a lot that has nothing to do with the logging feature

Reply:
A class that groups the s3 fields or gcp fields into individual objects would help achieve this.

Reply:
Good point. I did think about this but didn't think it was too confusing at the time.

I created a logging configuration interface that has a subset of the logging related config methods. The implementation to the logging configuration interface delgates to the configs interface so force a dependency to avoid multiple ways of configuration the same thing.